broadinstitute/pearl
# AR-101: adding mailing list functionality

This gives participants a way to sign up for a mailing list without reigstering for a study.  It also adds admin capabilities for viewing the mailing list, and updates populate files with some sample mailing list values.

On the participant side, right now the only entry to the mailing list is via the ineligible page.  Follow-on tickets will likely include supporting popping up the mailing list add component in a modal tied to navbar links and page buttons.

On the admin side, this brings in @tanstack/react-table, which I suspect we will want to use for all of our non-trivial admin-ui tables.  The admin functionality is bare-bones--follow-on tickets will likely include some sort of convenient copy of just the email addresses, and eventually built-in bulk emailing.  (the rows are currently selectable in preparation for those features, but those features haven't been added yet)

TO TEST:
1. restart admin and participant apps, repopulate ourhealth
2. in participant UI, click "join"
3. on the prereg survey, only check one of the options, and hit complete.
4. Confirm the ineligible screen now has a mailing list page.
![image](https://user-images.githubusercontent.com/2800795/227954033-e3191530-d22a-4be2-9844-9799914f4280.png)  
5. enter your name and email, and click "Join"
6. Confirm a success message appears
7. On the admin side, run npm install for ui-admin
8. go to https://localhost:3000/ourhealth/env/sandbox/mailingList
9. confirm you see the mailing list with an entry for the person you just added
![image](https://user-images.githubusercontent.com/2800795/227960448-a1696ca1-3d40-49e5-8b71-617c1baf9b60.png)
